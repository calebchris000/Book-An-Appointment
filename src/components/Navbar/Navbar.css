* {
  margin: 0;
  padding: 0;
}

.hamburger {
  width: 2rem;
  height: 2rem;
  position: fixed;
  float: left;
  margin: 1rem;
  z-index: 99;
  cursor: pointer;
  display: none;
}

.container {
    border: 1px solid #dbdbdb;
    background-color: #fefefe;
    float: left;
    position: fixed;
    top: 0;
    bottom: 0;
    z-index: 3;
    width: 14rem;
}

.navbar {
  height: 100vh;
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  /* gap: 1rem; */
  /* padding: 0.4rem; */
  margin-top: 6rem;
}

.img_container {
  display: flex;
  justify-content: center;
  align-items: center;
}

.logo {
  width: 8rem;
  transform: rotate(-20deg);
}

.links {
  list-style-type: none;
  /* border: 2px solid black; */
  padding: 1rem 1rem;
  color: black;
  font-weight: 700;
  text-decoration: none;
  margin-left: 0.3rem;
}

.links:hover {
  background-color: rgb(111, 185, 0);
  color: white;
}

@media (max-width: 768px) {
  .container  {
    display: none;
  }
  .hamburger {
    display: block;
  }
}
