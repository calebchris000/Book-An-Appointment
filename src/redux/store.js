import { configureStore } from "@reduxjs/toolkit";
import productDetailsReducer from "./productDetails/productDetailsSlice";
import productDescriptionReducer from "./productDescription/productDescriptionSlice";
import thunk from "redux-thunk";
import deleteCarReducer from './deleteCarSlice';

const store = configureStore({
  reducer: {
    productDetails: productDetailsReducer,
    productDescription: productDescriptionReducer,
  },
  middleware: [thunk],

  reducer: {
    deleteCar: deleteCarReducer,
  },
});

export default store;
