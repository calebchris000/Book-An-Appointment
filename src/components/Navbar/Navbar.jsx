import { Link } from "react-router-dom";
import Logo from "../../assets/logo.png";
import { GiHamburgerMenu } from "react-icons/gi";
import "./Navbar.css";
import { useEffect } from "react";
const Navbar = () => {
  useEffect(() => {
    window.addEventListener("resize", () => {
      const navbar = document.getElementsByClassName("container-nav")[0];
      if (window.innerWidth > 768) {
        navbar.style.display = "block";
      }
    });
  }, []);

  function handleClick(e) {
    document.querySelectorAll(".links").forEach((item) => {
      item.style.backgroundColor = "white";
      item.style.color = "black";
      item.classList.remove("selected");
    });
    // e.target.style.backgroundColor = "#6fb900";
    // e.target.style.color = "#fff";
    e.target.classList.add("selected");
  }

  function handleToggle() {
    const navbar = document.getElementsByClassName("container-nav")[0];
    navbar.style.display === "block" ? (navbar.style.display = "none") : (navbar.style.display = "block");
  }
  return (
    <>
      <div className="hamburger-container">
        <GiHamburgerMenu className="hamburger" onClick={handleToggle} />
      </div>
      <div className="container-nav">
        <div className="img_container">
          <img className="logo" src={Logo} alt="" />
        </div>
        <nav className="navbar">
          <Link className="links selected" to={"/"} onClick={handleClick}>
            HOME
          </Link>
          <Link className="links" to={"/reservation_form"} onClick={handleClick}>
            RESERVE
          </Link>
          <Link className="links" to={"/reservedCars"} onClick={handleClick}>
            MY RESERVATIONS
          </Link>
          <Link className="links" to={"/addcars"} onClick={handleClick}>
            ADD MOTORCYCLE
          </Link>
          <Link className="links" to={"/deleteCar"} onClick={handleClick}>
            DELETE
          </Link>
        </nav>
      </div>
    </>
  );
};

export default Navbar;
