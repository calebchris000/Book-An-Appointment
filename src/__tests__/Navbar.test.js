import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Navbar from '../components/Navbar/Navbar';


it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Navbar />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});