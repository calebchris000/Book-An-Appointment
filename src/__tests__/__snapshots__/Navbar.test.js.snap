// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="hamburger-container"
  >
    <svg
      className="hamburger"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
      />
    </svg>
  </div>,
  <div
    className="container-nav"
  >
    <div
      className="img_container"
    >
      <img
        alt=""
        className="logo"
        src="logo.png"
      />
    </div>
    <nav
      className="navbar"
    >
      <a
        className="links selected"
        href="/"
        onClick={[Function]}
      >
        HOME
      </a>
      <a
        className="links"
        href="/reservation_form"
        onClick={[Function]}
      >
        RESERVE
      </a>
      <a
        className="links"
        href="/reservations"
        onClick={[Function]}
      >
        MY RESERVATIONS
      </a>
      <a
        className="links"
        href="/addcars"
        onClick={[Function]}
      >
        ADD CAR
      </a>
      <a
        className="links"
        href="/deleteCar"
        onClick={[Function]}
      >
        DELETE
      </a>
    </nav>
  </div>,
]
`;
